src/Data/Generic.purs

```diff
@@ -222,6 +222,24 @@ data GenericSpine
   | SArray (Array (Unit -> GenericSpine))
   | SUnit
 
+-- | This instance is for debugging only.
+instance showGenericSpine :: Show GenericSpine
+    where
+        show  SUnit            = "SUnit"
+        show (SChar         c) = "SChar "    <> show  c
+        show (SString       s) = "SString "  <> show  s
+        show (SBoolean      b) = "SBoolean " <> show  b
+        show (SNumber       n) = "SNumber "  <> show  n
+        show (SInt          i) = "SInt "     <> show  i
+        show (SArray      arr) = "SArray "   <> show (map (\e -> e unit) arr)
+        show (SProd   lbl arr) = "SProd "    <> show  lbl <> " "
```

Comment:
This should show a function, something like `\_ -> ...`, so that the result is valid code which evaluates to the spine.